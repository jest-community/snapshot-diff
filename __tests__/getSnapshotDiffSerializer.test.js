// @flow

const snapshotDiff = require('../src/index');

expect.addSnapshotSerializer(snapshotDiff.getSnapshotDiffSerializer());

test('serialize text diff output', () => {
  expect(
    snapshotDiff(
      ['foo', 'bar', 'baz'].join('\n'),
      ['foo', 'baz', 'quoz'].join('\n')
    )
  ).toMatchSnapshot();
});

test('serialize array diff output', () => {
  expect(
    snapshotDiff(['foo', 'bar', 'baz'], ['foo', 'baz', 'quoz'])
  ).toMatchSnapshot();
});

test('serialize object diff output', () => {
  expect(snapshotDiff({ foo: 'bar' }, { foo: 'baz' })).toMatchSnapshot();
});
