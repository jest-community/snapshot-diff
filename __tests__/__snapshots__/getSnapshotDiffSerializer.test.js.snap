// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serialize array diff output 1`] = `
Snapshot Diff:
- First value
+ Second value

 Array [
   "foo",
-  "bar",
   "baz",
+  "quoz",
 ]
`;

exports[`serialize object diff output 1`] = `
Snapshot Diff:
- First value
+ Second value

 Object {
-  "foo": "bar",
+  "foo": "baz",
 }
`;

exports[`serialize text diff output 1`] = `
Snapshot Diff:
- First value
+ Second value

 foo
-bar
 baz
+quoz
`;
